Sources/Kitura/RouterResponse.swift

```diff
@@ -222,6 +222,19 @@ public class RouterResponse {
         }
         return self
     }
+    
+    /// Send an optional string.
+    ///
+    /// - Parameter str: the string to send.
+    /// - Returns: this RouterResponse.
+    @discardableResult
+    public func send(_ str: String?) -> RouterResponse {
+        guard !state.invokedEnd, let str = str else {
+            Log.warning("RouterResponse send(str:) invoked after end() for \(self.request.urlURL)")
```

Comment:
This doesn't look right, you've rolled two different checks into one guard, when they are checking two distinct cases:
- `state.invokedEnd` should log a warning ("send invoked after end") and return `self`,
- `str == nil` should log a different warning ("send invoked with a nil string") and return `send("")`.